lib/puppet_x/puppetlabs/cisco_ios/utility.rb

```diff
@@ -391,6 +391,11 @@ def self.convert_network_trunk_mode_modelled(trunk_mode_output)
       trunk_mode_output
     end
 
+    # Creates a deep copy of a given resource
+    def self.deep_copy(resource)
+      Marshal.load(Marshal.dump(resource))
```

Comment:
This doesn't need to be in a utility function. As it is, you use it once and you've written a unit test to test for it when it is a library function outside of your control.

Reply:
removed